### ISSUE TITLE
Loading methods and arguments

### ISSUE THREAD

**Issue by lhoestq** [huggingface/dataset-viewer#2414]
We can extend the `dataset-loading-tags` job to also store loading arguments that we can display on dataset pages in code snippets.

For example with the info to build the code snippet like
```json
{"loader": "webdataset", "url": "pipe:curl <hf url>"}
```
This way the front end can translate it to
```python
import webdadataset as wds

url = "pipe:curl <hf url>"
ds = wds.WebDataset(url).decode()
```

For croissant it would be
```json
{"loader": "mlcroissant", "url": "<croissant url>"}
```
for
```python
import mlcroissant as mlc

url = "<croissant url>"
ds = mlc.Dataset(url)
```

I will start with WebDataset and Croissant

**Comment by lhoestq** [comment]
I'm working on a pattern simplification/cleaning to have readable data files patterns to provide users e.g. in the "How to load in webdataset"

For example for timm/imagenet-1k-wds
Right now it looks like this 😬 
```python
{NamedSplit('train'): ['train[-._ 0-9/]**',
  '**/*[-._ 0-9/]train[-._ 0-9/]**',
  'training[-._ 0-9/]**',
  '**/*[-._ 0-9/]training[-._ 0-9/]**'],
 NamedSplit('validation'): ['validation[-._ 0-9/]**',
  '**/*[-._ 0-9/]validation[-._ 0-9/]**',
  'valid[-._ 0-9/]**',
  '**/*[-._ 0-9/]valid[-._ 0-9/]**',
  'dev[-._ 0-9/]**',
  '**/*[-._ 0-9/]dev[-._ 0-9/]**',
  'val[-._ 0-9/]**',
  '**/*[-._ 0-9/]val[-._ 0-9/]**']}
```
I'll aim for something like
```python
{NamedSplit('train'): 'imagenet1k-train-{{0000..1023}}.tar',
 NamedSplit('validation'): 'imagenet1k-validation-{{0000..1023}}.tar'}
```
Since WebDataset doesn't support "*" and only the above expansions

I also already have some code to simplify it to something like
```python
{NamedSplit('train'): '**/*-train-*.tar',
 NamedSplit('validation'): '**/*-validation-*.tar'}
```
Hopefully this is general enough so we can reuse it for json/csv/parquet in pandas/dask/spark etc 🙂

**Comment by severo** [comment]
closing, it's fixed with #2454
